### PR TITLE
Read IPFS stored profile avatars.

### PR DESCRIPTION
**Pull request by DanielVF**
Per issue #1795, we are working on moving away from storing the profile image data in the identity JSON. This PR takes care of the GraphQL side of this transition.

You can now fetch the avatarUrl from an identity GraphQL, even if that profile used an old style non-Url based avatar. In this case the avatar data in the profile is hashed by IPFS's splitting and hashing algorithms to create the actual IPFS url for the data.

In order that the avatar image will actually be at their IPFS URL, we also have a backfill script that works through the event history and for every non-deleted profile, uploads the avatar data to IPFS and checks that the expected IPFS hash and the actual uploaded hash match. It's single threaded for maximum obviousness, robustness, and debug-ability. (You could probably speed it up 30x by parallelizing the processing)

The listener should be storing the new `avatarUrl` field everywhere it stores, or webhooks, identities. In theory then, just rerunning the listener should update the database and ElasticSearch.

I've run it against Rinkeby, Mainnet, and Origin networks. In theory we should not need to run it again.

To run the backfill, go to the GraphQL package and `NETWORK=rinkeby npm run backfill_avatar`

### CODE REVIEW

**Comment by DanielVF** [comment]
Backfill stats:

Mainnet
```
{ 'SUCCESS: Avatar uploaded to IPFS': 4604,
  'SKIP: No avatar': 1269,
  'SKIP: Already using new avatarUrl': 349,
  'SKIP: Not actual data in data url for avatar': 4 }
```

Rinkeby
```
{ 'SUCCESS: Avatar uploaded to IPFS': 79,
  'SKIP: Not data url for avatar': 2,
  'SKIP: No avatar': 216,
  'SKIP: Already using new avatarUrl': 6 }
```

Origin:
```
{ 
'SKIP: No avatar': 8,
'SKIP: Already using new avatarUrl': 2
}
```

File: packages/graphql/src/resolvers/IdentityEvents.js [hunk 1]
```diff
@@ -105,6 +112,18 @@ export function identity({ id, ipfsHash }) {
       .filter(n => n)
       .join(' ')
 
+    // Make old style embedded avatars access by their IPFS hash.
+    if (identity.avatarUrl == undefined && identity.avatar != 'undefined') {
```
Comment (franckc) [inline packages/graphql/src/resolvers/IdentityEvents.js]: nit: use === and !=== operator to be consistent with the rest of our code base.

Same comment for the backfill script code.

**Comment by DanielVF** [comment]
Changed to use '===' and '!===' !

**Comment by franckc** [comment]
Thanks for fixing those nits !

**Pull request merged**